src/services/asset-discovery-service.ts

```diff
@@ -174,31 +217,44 @@ export default class AssetDiscoveryService extends PercyClientService {
       logger.debug(`Failed to load ${request.url()} : ${request.failure()!.errorText}}`)
     })
 
-    profile('--> assetDiscoveryService.page.goto', {url: rootResourceUrl})
-    await page.goto(rootResourceUrl)
-    profile('--> assetDiscoveryService.page.goto')
-
-    profile('--> assetDiscoveryService.waitForNetworkIdle')
-    await waitForNetworkIdle(page, this.networkIdleTimeout).catch(logError)
-    profile('--> assetDiscoveryService.waitForNetworkIdle')
+    try {
+      profile('--> assetDiscoveryService.page.goto', {url: rootResourceUrl})
+      await page.goto(rootResourceUrl)
+      profile('--> assetDiscoveryService.page.goto')
+
+      profile('--> assetDiscoveryService.waitForNetworkIdle')
+      await waitForNetworkIdle(page, this.networkIdleTimeout)
+      profile('--> assetDiscoveryService.waitForNetworkIdle')
+
+      profile('--> assetDiscoveryServer.waitForResourceProcessing')
+      const maybeResources: any[] = await Promise.all(maybeResourcePromises)
+      profile('--> assetDiscoveryServer.waitForResourceProcessing')
+
+      profile('--> assetDiscoveryService.pool.release', {url: rootResourceUrl})
+      await page.removeAllListeners('request')
+      await page.removeAllListeners('requestfinished')
+      await page.removeAllListeners('requestfailed')
+      await pool.release(page)
+      profile('--> assetDiscoveryService.pool.release')
+
+      return maybeResources.filter((maybeResource) => maybeResource != null)
+    } catch (error) {
```

Comment:
Love that we're going to catch these async errors now. Does it get rid of all of the deprecations?